realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -567,11 +601,32 @@ public void onChange() {
         realm.beginTransaction();
         realm.createObject(AllTypes.class);
         realm.commitTransaction();
-
-        assertEquals(1, counter.get());
-        assertEquals(1, realm.handlerController.weakChangeListeners.size());
     }
 
+    @Test
+    @RunTestInLooperThread
+    public void realmNotificationOrder() {
+        // Tests that global notifications are called in the order they are added
+        final AtomicBoolean firstListenerCalled = new AtomicBoolean(false);
+        Realm realm = looperThread.realm;
+        realm.addChangeListener(new RealmChangeListener() {
+
+            @Override
+            public void onChange() {
+                firstListenerCalled.set(true);
+            }
+        });
+        realm.addChangeListener(new RealmChangeListener() {
+            @Override
+            public void onChange() {
+                assertTrue(firstListenerCalled.get());
+                looperThread.testComplete();
+            }
+        });
+
+        realm.beginTransaction();
+        realm.commitTransaction();
```

Comment:
It could be accidentally pass the test if we only test `addListener(listener1)` `addListener(listener2)`, to avoid that we have to also test the other way around with same `listener1` and `listener2` : `addListener(listener2)` `addListener(listener1)`